bolt-modules/file/lib/puppet/functions/file/exists.rb

```diff
@@ -12,14 +12,31 @@
   #   file::exists('example/VERSION')
   dispatch :exists do
     scope_param
-    required_param 'String', :filename
+    required_param 'String[1]', :filename
     return_type 'Boolean'
   end
 
   def exists(scope, filename)
     # Send Analytics Report
-    Puppet.lookup(:bolt_executor) {}&.report_function_call(self.class.name)
-    found = Puppet::Parser::Files.find_file(filename, scope.compiler.environment)
+    executor = Puppet.lookup(:bolt_executor)
+    executor.report_function_call(self.class.name)
```

Comment:
Why this change? Don't functions available in apply blocks need to have the safe operator for the executor?

Reply:
Oh my gosh thank you for catching this! I had completely forgotten these could be called from inside apply blocks. Ended up reworking everything so that Bolt can locate these files without an executor.